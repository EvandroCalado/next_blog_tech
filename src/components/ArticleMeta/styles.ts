import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
