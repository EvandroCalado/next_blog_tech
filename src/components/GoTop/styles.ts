import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    position: fixed;
    bottom: 3rem;
    right: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    text-decoration: none;
    transition: all 300ms ease-in-out;

    &:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
