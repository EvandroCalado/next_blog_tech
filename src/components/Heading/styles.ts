import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: 900;
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: 900;
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: 900;
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    font-weight: 900;
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primary : theme.colors.white};
    line-height: 1.2;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
    text-align: center;
  `}
`;
