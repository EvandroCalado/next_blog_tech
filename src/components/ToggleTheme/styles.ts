import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  position: absolute;
  right: 5rem;
  top: 3.1rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.7rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: transparent;
`;

export const Input = styled.input`
  ${({ theme }) => css`
    display: none;

    &:checked + ${Slider} {
      background-color: ${theme.colors.secondary};
    }

    &:checked + ${Slider}::before {
      transform: translateX(2.5rem);
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    border: 1px solid ${theme.colors.primary};
    background-color: ${theme.colors.darkerGray};
    transition: all 300ms ease-in-out;

    &::before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.2rem;
      bottom: 0.2rem;
      background-color: ${theme.colors.white};
      transition: all 300ms ease-in-out;
    }
  `}
`;
