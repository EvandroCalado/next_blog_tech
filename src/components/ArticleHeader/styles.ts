import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
