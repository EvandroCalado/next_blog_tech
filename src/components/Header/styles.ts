import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.dark};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${Title} {
      margin: ${theme.spacings.small} 0;
      text-align: left;
    }

    @media ${theme.media.medium} {
      flex-flow: column wrap;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.medium} {
      margin: 0;
    }
  `}
`;
