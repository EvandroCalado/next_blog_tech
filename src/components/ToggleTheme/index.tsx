import { useContext, useEffect, useState } from 'react';
import { BlogThemeContext } from '../../contexts/BlogThemeContext';
import * as Styled from './styles';

const ToggleTheme = () => {
  const { setTheme } = useContext(BlogThemeContext);
  const [cheked, setChecked] = useState(false);

  useEffect(() => {
    const localTheme = localStorage.getItem('theme');
    if (!localTheme) return;
    const newTheme = JSON.parse(localTheme);

    if (newTheme.name === 'inverted') {
      setChecked(true);
    }
  }, []);

  useEffect(() => {
    setTheme(cheked ? 'inverted' : 'default');
  }, [cheked, setTheme]);

  const handleChange = () => {
    setChecked(!cheked);
    setTheme(cheked ? 'inverted' : 'default');
  };

  return (
    <Styled.Wrapper>
      <Styled.Label>
        Toggle light and dark modes
        <Styled.Input
          type="checkbox"
          value="Dark mode active"
          onChange={handleChange}
          checked={cheked}
        />
        <Styled.Slider></Styled.Slider>
      </Styled.Label>
    </Styled.Wrapper>
  );
};

export default ToggleTheme;
