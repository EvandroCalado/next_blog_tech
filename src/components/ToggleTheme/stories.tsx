import ToggleTheme from '.';
import { Meta, Story } from '@storybook/react/types-6-0';

export default {
  title: 'ToggleTheme',
  component: ToggleTheme,
} as Meta;

export const Template: Story = (args) => (
  <div>
    <ToggleTheme {...args} />
  </div>
);
