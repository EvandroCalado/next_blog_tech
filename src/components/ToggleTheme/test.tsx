import ToggleTheme from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

describe('<ToggleTheme />', () => {
  it('should render', () => {
    renderTheme(<ToggleTheme />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
