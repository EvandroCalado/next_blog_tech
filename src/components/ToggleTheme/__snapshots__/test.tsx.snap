// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ToggleTheme /> should render input and toggle betwen theme when toggling input 1`
] = `
.c0 {
  position: absolute;
  right: 1rem;
  top: 3.1rem;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

.c0 > svg {
  width: 2.5rem;
  color: #fff;
}

.c2 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.7rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: transparent;
}

.c3 {
  display: none;
}

.c3:checked + .c4 {
  background-color: #dc143c;
}

.c3:checked + .c4::before {
  -webkit-transform: translateX(2.5rem);
  -ms-transform: translateX(2.5rem);
  transform: translateX(2.5rem);
}

.c5 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border: 1px solid #fff;
  background-color: #ccc;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c5::before {
  content: '';
  position: absolute;
  height: 2.1rem;
  width: 2.1rem;
  left: 0.2rem;
  bottom: 0.2rem;
  background-color: #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<div>
  <div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 9c1.65 0 3 1.35 3 3s-1.35 3-3 3-3-1.35-3-3 1.35-3 3-3m0-2c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 0 0 0-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 0 0 0-1.41.996.996 0 0 0-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 0 0 0-1.41.996.996 0 0 0-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
      />
    </svg>
    <label
      class="c2"
    >
      Toggle light and dark modes
      <input
        aria-label="Checkbox"
        class="c3"
        type="checkbox"
        value="Dark mode active"
      />
      <span
        class="c4 c5"
      />
    </label>
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.37 5.51A7.35 7.35 0 0 0 9.1 7.5c0 4.08 3.32 7.4 7.4 7.4.68 0 1.35-.09 1.99-.27A7.014 7.014 0 0 1 12 19c-3.86 0-7-3.14-7-7 0-2.93 1.81-5.45 4.37-6.49zM12 3a9 9 0 1 0 9 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 0 1-4.4 2.26 5.403 5.403 0 0 1-3.14-9.8c-.44-.06-.9-.1-1.36-.1z"
      />
    </svg>
  </div>
</div>
`;
